File: CPP Codes/Stack.cpp
#include <iostream>
#include <vector>

using namespace std;

class Stack {
private:
    vector<int> elements;

public:
    // Push: Adds an element to the top of the stack
    void push(int data) {
        elements.push_back(data);
    }

    // Pop: Removes and returns the element at the top of the stack
    int pop() {
        if (elements.empty()) {
            cout << "Stack is empty. Cannot pop." << endl;
            return -1; // Indicating an error
        }
        int topElement = elements.back();
        elements.pop_back();
        return topElement;
    }

    // Peek: Returns the element at the top of the stack without removing it
    int peek() {
        if (elements.empty()) {
            cout << "Stack is empty. Cannot peek." << endl;
            return -1; // Indicating an error
        }
        return elements.back();
    }

    // Size: Returns the number of elements in the stack
    int size() const {
        return elements.size();
    }

    // IsEmpty: Checks if the stack is empty
    bool isEmpty() const {
        return elements.empty();
    }
};

int main() {
    Stack stack;
    int choice, value;

    do {
        cout << "\nStack Operations Menu:\n";
        cout << "1. Push\n";
        cout << "2. Pop\n";
        cout << "3. Peek\n";
        cout << "4. Size\n";
        cout << "5. Is Empty\n";
        cout << "6. Exit\n";
        cout << "Enter your choice: ";
        cin >> choice;

        switch (choice) {
            case 1:
                cout << "Enter value to push: ";
                cin >> value;
                stack.push(value);
                cout << "Pushed " << value << " onto the stack.\n";
                break;
            case 2:
                value = stack.pop();
                if (value != -1) {
                    cout << "Popped " << value << " from the stack.\n";
                }
                break;
            case 3:
                value = stack.peek();
                if (value != -1) {
                    cout << "Top element is " << value << ".\n";
                }
                break;
            case 4:
                cout << "Current stack size: " << stack.size() << endl;
                break;
            case 5:
                cout << "Is stack empty? " << (stack.isEmpty() ? "Yes" : "No") << endl;
                break;
            case 6:
                cout << "Exiting...\n";
                break;
            default:
                cout << "Invalid choice. Please try again.\n";
        }
    } while (choice != 6);

    return 0;
}

File: JAVA Codes/Stack.java
import java.util.Scanner;

class Stack {
    private int[] elements;
    private int size;
    private int capacity;

    public Stack(int capacity) {
        this.capacity = capacity;
        this.elements = new int[capacity];
        this.size = 0;
    }

    // Push: Adds an element to the top of the stack
    public void push(int data) {
        if (size == capacity) {
            // Resize the stack if it is full
            capacity *= 2;
            int[] newElements = new int[capacity];
            System.arraycopy(elements, 0, newElements, 0, size);
            elements = newElements;
        }
        elements[size++] = data;
    }

    // Pop: Removes and returns the element at the top of the stack
    public int pop() {
        if (size == 0) {
            System.out.println("Stack is empty. Cannot pop.");
            return -1; // Indicating an error
        }
        return elements[--size];
    }

    // Peek: Returns the element at the top of the stack without removing it
    public int peek() {
        if (size == 0) {
            System.out.println("Stack is empty. Cannot peek.");
            return -1; // Indicating an error
        }
        return elements[size - 1];
    }

    // Size: Returns the number of elements in the stack
    public int size() {
        return size;
    }

    // IsEmpty: Checks if the stack is empty
    public boolean isEmpty() {
        return size == 0;
    }
}

public class Main {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        Stack stack = new Stack(2); // Initial capacity of 2

        int choice, value;

        do {
            System.out.println("\nStack Operations Menu:");
            System.out.println("1. Push");
            System.out.println("2. Pop");
            System.out.println("3. Peek");
            System.out.println("4. Size");
            System.out.println("5. Is Empty");
            System.out.println("6. Exit");
            System.out.print("Enter your choice: ");
            choice = scanner.nextInt();

            switch (choice) {
                case 1:
                    System.out.print("Enter value to push: ");
                    value = scanner.nextInt();
                    stack.push(value);
                    System.out.println("Pushed " + value + " onto the stack.");
                    break;
                case 2:
                    value = stack.pop();
                    if (value != -1) {
                        System.out.println("Popped " + value + " from the stack.");
                    }
                    break;
                case 3:
                    value = stack.peek();
                    if (value != -1) {
                        System.out.println("Top element is " + value + ".");
                    }
                    break;
                case 4:
                    System.out.println("Current stack size: " + stack.size());
                    break;
                case 5:
                    System.out.println("Is stack empty? " + (stack.isEmpty() ? "Yes" : "No"));
                    break;
                case 6:
                    System.out.println("Exiting...");
                    break;
                default:
                    System.out.println("Invalid choice. Please try again.");
            }
        } while (choice != 6);

        scanner.close();
    }
}
